test/TestRunner.py

```diff
@@ -17,121 +17,109 @@
+
+def test_unicode_parseable_colored_formatting(default_rules_collection):
+    filename = 'test/unicode.yml'
+    runner = Runner(default_rules_collection, filename, [], [], [])
+    matches = runner.run()
+    formatter = ansiblelint.formatters.ParseableFormatter(os.getcwd(), True)
+    formatter.format(matches[0], colored=True)
+
+
+def test_unicode_quiet_colored_formatting(default_rules_collection):
+    filename = 'test/unicode.yml'
+    runner = Runner(default_rules_collection, filename, [], [], [])
+    matches = runner.run()
+    formatter = ansiblelint.formatters.QuietFormatter(os.getcwd(), True)
+    formatter.format(matches[0], colored=True)
+
+
+def test_unicode_standard_color_formatting(default_rules_collection):
+    filename = 'test/unicode.yml'
+    runner = Runner(default_rules_collection, filename, [], [], [])
+    matches = runner.run()
+    formatter = ansiblelint.formatters.Formatter(os.getcwd(), True)
+    formatter.format(matches[0], colored=True)
+
+
+def test_runner_excludes_paths(default_rules_collection):
+    filename = abspath('examples/lots_of_warnings.yml', os.getcwd())
+    excludes = [filename]
+    runner = Runner(default_rules_collection, filename, [], [], excludes)
+    assert (len(runner.run()) == 0)
+
+
+def test_runner_block_count(default_rules_collection):
+    filename = 'test/block.yml'
+    runner = Runner(default_rules_collection, filename, [], [], [])
+    assert (len(runner.run()) == 0)
+
+
+def test_runner_become_count(default_rules_collection):
+    filename = 'test/become.yml'
+    runner = Runner(default_rules_collection, filename, [], [], [])
+    assert (len(runner.run()) == 0)
+
+
+def test_runner_empty_tags_count(default_rules_collection):
+    filename = 'test/emptytags.yml'
+    runner = Runner(default_rules_collection, filename, [], [], [])
+    assert (len(runner.run()) == 0)
+
+
+def test_runner_encrypted_secrets(default_rules_collection):
+    filename = 'test/contains_secrets.yml'
+    runner = Runner(default_rules_collection, filename, [], [], [])
+    assert (len(runner.run()) == 0)
+
+
+def test_dir_with_trailing_slash(default_rules_collection):
+    filename = 'test/'
+    runner = Runner(default_rules_collection, filename, [], [], [])
+    assert (list(runner.playbooks)[0][1] == 'role')
+
+
+def test_dir_with_fullpath(default_rules_collection):
+    filename = os.path.abspath('test')
+    runner = Runner(default_rules_collection, filename, [], [], [])
+    assert (list(runner.playbooks)[0][1] == 'role')
+
+
+def test_files_not_scanned_twice(default_rules_collection):
+    checked_files = set()
+
+    filename = os.path.abspath('test/common-include-1.yml')
+    runner = Runner(default_rules_collection, filename, [], [], [], 0, checked_files)
+    run1 = runner.run()
+
+    filename = os.path.abspath('test/common-include-2.yml')
+    runner = Runner(default_rules_collection, filename, [], [], [], 0, checked_files)
+    run2 = runner.run()
+
+    assert ((len(run1) + len(run2)) == 1)
```

Comment:
note: test in this module are also parametrizable but this can be postponed.

Reply:
This is now three parametrized tests (but the last that stands on its own).